### PR TITLE
Decorated-person-class

### PR DESCRIPTION
**Pull request by Obimbo07**
This pull request introduces the following features 

- Create a class Nameable.

    - Implement a method called correct_name that will raise a NotImplementedError.
    - Turn your Person class to Nameable
    - Make sure that your Person class inherits from Nameable
    - Make sure that this class has a method correct_name implemented. It should simply return the name attribute.

- Prepare base Decorator

        - Make sure that it inherits from Nameable.
        - In the constructor assign a nameable object from params to an instance variable.
        - Implemented the correct_name method that returns the result of the correct_name method of the @nameable.
        - Prepare CapitalizeDecorator and TrimmerDecorator
        - For the CapitalizeDecorator:
        - Create a class that inherits from the base Decorator class.
        - Implement a method correct_name that capitalizes the output of @nameable.correct_name.
For the TrimmerDecorator:
Create a class that inherits from the base Decorator class.
Implemented a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,3 +1,34 @@
...
+end
+
+class TrimmerDecorator < NameableDecorator
+  def correct_name
+    trimmed_name = @nameable.correct_name[0, 10]
+    trimmed_name.sub!(/\s\w*\z/, '')
+    trimmed_name
+  end
+end
+
```
Comment (Gambit142) [inline person.rb]: - [ ] Please ensure that all classes are put in their files. It is a normal convention and clean way to write codes.  It makes it easy to predict the location of classes and their methods and helps in debugging and preventing unpredictable behaviour in your application. Kindly implement this change. 👍 

#### N.B: After separating the classes in each file, kindly test the code and ensure they work properly. You might have to require some files, so you can use `require_relative` to handle that. 💻 😎

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Obimbo07 👋, 

Good job so far!

## Highlights
- Setup linters correctly ✔️ 
- Created `Nameable` class ✔️
- Created a based decorator class ✔️ 
- Created other decorators ✔️  

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: person.rb (hunk 1 above)
Comment (Obimbo07) [reply person.rb]: Resolved

Comment (prolajumokeoni) [reply person.rb]: Hi @Obimbo07,  Good job so far, after separating the classes did you run the code below?
```

person = Person.new(22, 'maximilianus')
 person.correct_name
 capitalizedPerson = CapitalizeDecorator.new(person)
 capitalizedPerson.correct_name
 capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
 capitalizedTrimmedPerson.correct_name

```

Comment (Obimbo07) [reply person.rb]: yes i have done so

Comment (Obimbo07) [reply person.rb]: Fixed the conflicting codes

Comment (Obimbo07) [reply person.rb]: i have Fixed the conflicting codes

**Review by prolajumokeoni — APPROVED** [review]
Hi @Obimbo07 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - Class is decorated ✅ 
  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**